src/app/view/perspectives/post-it/post-it-perspective.component.ts

```diff
@@ -136,122 +136,134 @@ export class PostItPerspectiveComponent implements OnInit, OnDestroy {
     this.pageSubscriptions.forEach(subscription => subscription.unsubscribe());
   }
 
+  public usedCollections(): CollectionModel[] {
+    return Array.from(new Set(this.postIts.map(postIt => postIt.document.collection)));
+  }
+
+  public fetchQueryDocuments(queryModel: QueryModel): void {
+    this.store.dispatch(new DocumentsAction.Get({query: queryModel}));
+  }
+
+  public isAddButtonShown(): boolean {
+    return this.editable && this.navigationHelper.hasOneCollection();
+  }
+
+  private checkAllLoaded(documents: DocumentModel[]): void {
+    this.allLoaded = documents.length === 0;
+  }
+
+  public ngOnDestroy(): void {
+    this.deletionHelper.destroy();
+    this.navigationHelper.destroy();
+    this.infiniteScroll.destroy();
+    this.pageSubscriptions.forEach(subscription => subscription.unsubscribe());
```

Comment:
We should check if these fields are initialized before destroying them. Otherwise, if there is an error on initialization, `ngOnDestroy` will throw another error.